### ISSUE TITLE
Error in rule bindetect

### ISSUE THREAD

**Issue by VenkateshLab** [loosolab/TOBIAS#181]
Hi, I using Snakemake for analysis my ATAC seq data. im facing a issue in bindetect using the snake make. 

```
Error in rule bindetect:
    jobid: 10
    input: P0E13/motifs/all_motifs.txt, P0E13/footprinting/Postnatal_footprints.bw, P0E13/footprinting/Embroyo_footprints.bw, P0E13/flatfiles/mm10.fa, Ref/peaks.bed, P0E13/peak_annotation/peaks_header.txt
    output: P0E13/TFBS
    log: P0E13/logs/bindetect.log (check log file(s) for error details)
    conda-env: /media/volume/sdf/Manoj/TOBIAS/Test/TOBIAS_snakemake/Temp/3e6631153f293e8d599565ec858a7ef1_
    shell:
        TOBIAS BINDetect --motifs P0E13/motifs/all_motifs.txt --signals P0E13/footprinting/Postnatal_footprints.bw P0E13/footprinting/Embroyo_footprints.bw --genome P0E13/flatfiles/mm10.fa --peaks Ref/peaks.bed --peak_header P0E13/peak_annotation/peaks_header.txt --cores 25 --outdir P0E13/TFBS --cond_names Postnatal Embroyo  &> P0E13/logs/bindetect.log; mkdir -p P0E13/overview;cp P0E13/TFBS/*.txt P0E13/overview;cp P0E13/TFBS/*.xlsx P0E13/overview;cp P0E13/TFBS/*.pdf P0E13/overview;
        (one of the commands exited with non-zero exit code; note that snakemake uses bash strict mode!)

Removing output files of failed job bindetect since they might be corrupted:
P0E13/TFBS
Exiting because a job execution failed. Look above for error message
Complete log: .snakemake/log/2023-01-13T063728.641108.snakemake.log
```

log file of bind detect 
[bindetect.log](https://github.com/loosolab/TOBIAS/files/10409033/bindetect.log)

config.yaml file.
```
-------------------------------------------------------------------------#
#-------------------------- TOBIAS input data ----------------------------#
#-------------------------------------------------------------------------#

data:
  Postnatal: [ATAC/P0.bam]                                   #list of .bam-files
  Embroyo: [ATAC/E13.bam]  #list of .bam-files
  
run_info:
  organism: mouse                           #mouse/human/zebrafish
  fasta: Ref/mm10.fa             #.fasta-file containing organism genome. NOTE: must be uncompressed .fa or bgzip compressed compatible with samtools
  blacklist: Ref/blacklist.bed        #.bed-file containing blacklisted regions
  gtf: Ref/mm10.gtf                  #.gtf-file for annotation of peaks. NOTE: must be uncompressed .gtf
  motifs: Ref/jaspar/*          #motifs (directory with files or individual files in MEME/JASPAR/PFM format)
  output: P0E13                       #output directory 
  peaks: Ref/peaks.bed          #optional; pre-calculated annotated peaks
  #peaks_header: data/annotated_peaks_header.txt #optional; header for pre-calculated annotated peaks

#Flags for parts of pipeline to include/exclude (all are True by default)
flags:
  plot_comparison: True #True/False
  plot_correction: True
  plot_venn: True
  coverage: True
  wilson: True

#-------------------------------------------------------------------------#
#----------------------- Default module parameters -----------------------#
#-------------------------------------------------------------------------#

macs: "--nomodel --shift -100 --extsize 200 --broad"
atacorrect: ""
footprinting: ""
bindetect: ""

```

my data were donwloaded from encodeproject 
P0 - https://www.encodeproject.org/experiments/ENCSR310MLB/
E13 - https://www.encodeproject.org/experiments/ENCSR903GMO/

**Comment by msbentsen** [comment]
Hi @VenkateshLab ,
Thank you for sharing the log-file, that is really helpful! The error is already well-known as described here: 
https://github.com/loosolab/TOBIAS/wiki/FAQ#5-i-get-an-error-stating-too-many-files-open---what-do-i-do

Please adjust the maximum number of open filehandles on your system by:
`$ ulimit -n 3000`

Hope that solves the issue.

**Comment by VenkateshLab** [comment]
@msbentsen thanks it fixed that, but now im getting an new error 
![image](https://user-images.githubusercontent.com/118877790/212327584-ea15a158-d3d0-408a-8119-7fdaa4ca0ad1.png)
```
Error in rule bindetect:
    jobid: 10
    input: P0E13/motifs/all_motifs.txt, P0E13/footprinting/Postnatal_footprints.bw, P0E13/footprinting/Embroyo_footprints.bw, P0E13/flatfiles/mm10.fa, Ref/peaks.bed, P0E13/peak_annotation/peaks_header.txt
    output: P0E13/TFBS
    log: P0E13/logs/bindetect.log (check log file(s) for error details)
    conda-env: /media/volume/sdf/Manoj/TOBIAS/Test/TOBIAS_snakemake/Temp/3e6631153f293e8d599565ec858a7ef1_
    shell:
        TOBIAS BINDetect --motifs P0E13/motifs/all_motifs.txt --signals P0E13/footprinting/Postnatal_footprints.bw P0E13/footprinting/Embroyo_footprints.bw --genome P0E13/flatfiles/mm10.fa --peaks Ref/peaks.bed --peak_header P0E13/peak_annotation/peaks_header.txt --cores 25 --outdir P0E13/TFBS --cond_names Postnatal Embroyo  &> P0E13/logs/bindetect.log; mkdir -p P0E13/overview;cp P0E13/TFBS/*.txt P0E13/overview;cp P0E13/TFBS/*.xlsx P0E13/overview;cp P0E13/TFBS/*.pdf P0E13/overview;
        (one of the commands exited with non-zero exit code; note that snakemake uses bash strict mode!)

Removing output files of failed job bindetect since they might be corrupted:
P0E13/TFBS
Exiting because a job execution failed. Look above for error message
Complete log: .snakemake/log/2023-01-13T122136.013369.snakemake.log
```
i have also encloed the log file with this.
[bindetect (1).log](https://github.com/loosolab/TOBIAS/files/10411809/bindetect.1.log)

**Comment by msbentsen** [comment]
Thank you for sending the log. I see, this arises because of the large number of input motifs. I would probably look at reducing the number of motifs you are using, as 2000+ motifs will make a very full plot. JASPAR has motif subsets specific for vertebrates, which might be better suited for mouse data. So if you reduce the number of motifs the error will also go away.

But regardless, this is definitely a bug! I just released `TOBIAS 0.15.1` to solve this issue if you still want to use 2000+ motifs. To use the new version in the pipeline, you might specify the version in the environment file here:
https://github.molgen.mpg.de/loosolab/TOBIAS_snakemake/blob/master/environments/tobias.yaml#L12-L13
as 
```
 - pip:
    - tobias==0.15.1
```
Or just make sure the conda environments are installed again during the run. Hope this solves the issue!

**Comment by VenkateshLab** [comment]
@msbentsen  Thank you, I have used motifs specific to vertebrates which helped to solve the problem. I will be testing with the recent version of `TOBIAS (0.15.1)` and will let you know if I encounter any errors while using it with more than 2000 motifs.
